docs/deployment_design/design.md

```diff
@@ -0,0 +1,63 @@
+# Llama index deployment design
+
+## Component diagram
+
+![Alt text](components.png)
+
+## Deployment diagram
+
+![Alt text](deployment.png)
+
+## LlamaIndex API startup sequence diagram
+
+```mermaid
+sequenceDiagram
+    participant creator as App Creator
+
+    create participant sc as :StorageContext
+    creator->>sc: create with index_folder
+    create participant index as :BaseIndex
+    creator->>index: create with storage_context
+    create participant retriever as :VectorIndexRetriever
+    creator->>retriever: create with index
+    create participant rqe as :RetrieverQueryEngine
+    creator->>rqe: create with retriever
+    create participant conf as :Config
+    creator->>conf: create with query_engine
+    create participant app as :Flask
+    creator->>app: create with config
+
+```
+
+**Resources:**
+
+- [Loading data in LlamaIndex from
+  filesystem](https://docs.llamaindex.ai/en/stable/module_guides/storing/save_load.html#loading-data)
+
+## LLamaIndex API search sequence diagram
+
+```mermaid
+sequenceDiagram
+    actor user
+    participant app as app:Flask
+    participant config as config:Config
+    participant rqe as :RetrieverQueryEngine
+
+    user->>+app: POST /search query_string
+    app->>config: get query_engine
+    app->>+rqe: query with query_string
+    rqe-->>-app: response
+    app->app: generate results from response.nodes
+    app-->>-user: results
+```
+
+**Resources:**
+
+- [Creating a flask API with
+  LlamaIndex](https://docs.llamaindex.ai/en/stable/understanding/putting_it_all_together/apps/fullstack_app_guide.html#flask-backend)
+
+**Note:**
+
+The retriever query potentially takes some time (more than 5 seconds in some
+instances). This could be problematic as the user expects near instant results.
+We might have to optimize this process.
```

Comment:
By optimizing the process, do you have in mind ideas like cutting CHATGPT from being part of the query response and other ideas you mentioned during the backend meeting? If so, I would encourage you to add those optimization ideas to your documentation.

Reply:
Sure thing. I'm coordinating with @leejaeka on this.